WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/builders/BackupDownloadStateListItemBuilder.kt

```diff
@@ -229,4 +232,44 @@ class BackupDownloadStateListItemBuilder @Inject constructor(
             label = UiStringRes(labelRes),
             itemBottomMarginResId = itemBottomMarginResId
     )
+
+    fun updateCheckboxes(
+        uiState: BackupDownloadUiState,
+        itemType: JetpackAvailableItemType
+    ): List<JetpackListItemState> {
+        var checkedCount = 0
```

Comment:
💡 This code took me a while to understand. We might want to consider separating the "change state" logic and "count" logic. Perhaps something like this might work:
```
        val updateListItems = uiState.items.map { state ->
                if (state is CheckboxState && state.availableItemType == itemType) {
                    state.copy(checked = !state.checked)
                } else {
                    state
                }
        }
        val atLeastOneChecked = updateListItems.filterIsInstance<CheckboxState>().find { it.checked } != null
        return updateDetailsActionButtonState(updateListItems, enableActionButton = atLeastOneChecked)
```

Wdyt?

Reply:
Nice upgrade. In commit 14d3a34